ta/avb/include/ta_avb.h

```diff
@@ -45,4 +45,20 @@
  */
 #define TA_AVB_CMD_WRITE_LOCK_STATE	3
 
+/*
+ * Reads a persistent value corresponding to the given name.
+ *
+ * in		params[0].memref:	persistent value name
+ * inout	params[1].memref:	read persistent value buffer
```

Comment:
Why is this inout?

Reply:
@jenswi-linaro https://android.googlesource.com/platform/external/avb/+/master/libavb/avb_ops.h#284

```

AvbIOResult (*read_persistent_value)(AvbOps* ops,
  const char* name,
  size_t buffer_size,
  uint8_t* out_buffer,
  size_t* out_num_bytes_read);
```
We need to return out_num_bytes_read, I'm using `params[1].memref.size` ( I doubt that in this case is good idea, but I just wanted to avoid introducing additional param for this ("params[2].value.a", for example))

Reply:
"out" should be enough then.

Reply:
Yes, that's how out buffers works. You supply a pointer to the buffer and the size. When the buffer is filled in the TA can change the size to reflect how much data was copied.

Reply:
@jenswi-linaro got it, thanks for the explanation